### PR TITLE
Build time optimizations

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

Adds the possibility to glob together all tests into one big cpp file.

Timings (debug mode, clang10, target=runTests):

|          | noglob | glob| 
| ------ | -------------- | ----------- | 
-j 1      | 8:50 | 4:22 |
-j10     | 2:55 | 3:58 |

Can be enabled with option `AUTOPAS_USE_TEST_GLOB`

This PR adds -DAUTOPAS_USE_TEST_GLOB=On to relevant jenkins calls to cmake.

### CODE REVIEW

File: tests/testAutopas/addNameSpace.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (FG-TUM) [inline tests/testAutopas/addNameSpace.sh:1]: add some context for our future selfs what this script can / should be used for

Comment (FG-TUM) [reply tests/testAutopas/addNameSpace.sh:1]: also please mention somewhere that all tests should have their own namespace (probably contribution guidelines)

File: tests/testAutopas/CMakeLists.txt [hunk 2]
```diff
@@ -13,7 +13,19 @@ if (AUTOPAS_ENABLE_CUDA)
...
-add_executable(runTests $<$<BOOL:${AUTOPAS_ENABLE_CUDA}>:${CU_SRC}> ${MY_SRC})
+option(AUTOPAS_USE_TEST_GLOB "Testing: Use one cpp file that includes all other cpp files. Reduces the amount of needed template instantiations." ON)
+
+if(AUTOPAS_USE_TEST_GLOB)
+    set(NEW_LIST ${MY_SRC})
+    list(TRANSFORM NEW_LIST PREPEND "#include \"")
+    list(TRANSFORM NEW_LIST APPEND "\"\n")
+    set(AUTOPAS_TEST_FILES ${CMAKE_CURRENT_BINARY_DIR}/test_glob.cpp)
+    file(WRITE ${AUTOPAS_TEST_FILES} ${NEW_LIST})
+else()
```
Comment (FG-TUM) [inline tests/testAutopas/CMakeLists.txt:24]: This literally creates a file with the content 
```
#include "<TestA.cpp>"
#include "<TestB.cpp>"
...
``` 
right? Please comment this. Especially that transform changes each member of the list.

Comment (SteffenSeckler) [reply tests/testAutopas/CMakeLists.txt]: done.

File: tests/testAutopas/addNameSpace.sh (hunk 1 above)
Comment (SteffenSeckler) [reply tests/testAutopas/addNameSpace.sh:1]: done

File: Jenkinsfile [hunk 3]
```diff
@@ -104,7 +104,7 @@ pipeline {
                     steps {
                         container('cuda-10') {
                             dir("build-cuda") {
-                                sh "cmake -DAUTOPAS_ENABLE_CUDA=ON -DCCACHE=ON .."
+                                sh "cmake -DAUTOPAS_USE_TEST_GLOB=On -DAUTOPAS_ENABLE_CUDA=ON -DCCACHE=ON .."
```
Comment (FG-TUM) [inline Jenkinsfile:107]:
```suggestion
                                sh "cmake -DAUTOPAS_USE_TEST_GLOB=ON -DAUTOPAS_ENABLE_CUDA=ON -DCCACHE=ON .."
```
This triggers me way more than it should ...
Btw you spelled all new flags that way.

**Comment by SteffenSeckler** [comment]
closed in favor of #568

**Pull request closed without merging**